custom/abdm/discovery_request.py

```diff
@@ -0,0 +1,136 @@
+from corehq.form_processor.models import CommCareCase
+from corehq.toggles import ABDM_INTEGRATION
+from corehq.toggles.shortcuts import find_domains_with_toggle_enabled
+from corehq.apps.es.case_search import CaseSearchES, case_property_range_query, sounds_like_text_query
+from corehq.apps.es import queries
+
+from abdm_integrator.hip.exceptions import DiscoveryNoPatientFoundError, DiscoveryMultiplePatientsFoundError
+
+from datetime import date
+from abdm_integrator.const import IdentifierType
+
+PATIENT_CASE_TYPE = 'abdm_patient_discovery_data'
+HIP_ID_PROPERTY = 'hip_id'
+HEALTH_ID_PROPERTY = 'abha_address'
+PHONE_NUMBER_PROPERTY = 'phone_number'
+DATE_OF_BIRTH_PROPERTY = 'dob'
+GENDER_PROPERTY = 'sex'
+ABHA_NUMBER_PROPERTY = 'abha_id'
+NAME_PROPERTY = 'case_name'
+
+CARE_CONTEXT_CASE_TYPE = 'fhir_care_context'
+YEARS_RANGE_TO_MATCH = 2
+
+
+def base_discover_query(domains, hip_id):
+    return CaseSearchES().filter(
+        {
+            'terms': {
+                'domain.exact': domains
+            }
+        }
+    ).case_type(PATIENT_CASE_TYPE).case_property_query(HIP_ID_PROPERTY, hip_id).sort('opened_on')
+
+
+def discover_by_health_id(patient_details, hip_id, abdm_enabled_domains):
+    health_id_query = base_discover_query(abdm_enabled_domains, hip_id).case_property_query(
+        HEALTH_ID_PROPERTY,
+        patient_details.health_id
+    )
+    return health_id_query.values('name', 'domain', 'indices.referenced_id')
+
+
+def discover_by_demographics(patient_details, hip_id, abdm_enabled_domains):
```

Comment:
Consider adding tests for these ES query methods

Reply:
Added test cases 24a304259b58cfedeb0dd07ee23276c2a2d68959